extensions/amp-story/1.0/amp-story.js

```diff
@@ -452,11 +464,54 @@ export class AmpStory extends AMP.BaseElement {
       );
     }
 
+    this.applyDesktopAspectRatioAttribute_();
+
     if (this.maybeLoadStoryDevTools_()) {
       return;
     }
   }
 
+  /**
+   * Grab the desktop-aspect-ratio attribute, clamp the value
+   * between 1/2 and 3/4 aspect ratios, and apply it to the root element.
+   * @private
+   */
+  applyDesktopAspectRatioAttribute_() {
+    if (
+      this.isLandscapeSupported_() ||
+      !this.element.hasAttribute('desktop-aspect-ratio')
+    ) {
+      return;
+    }
+
+    const splittedRatio = this.element
+      .getAttribute('desktop-aspect-ratio')
+      .split(':');
+    if (splittedRatio.length != 2 || splittedRatio[1] == 0) {
+      return;
+    }
+
+    const desktopAspectRatio = Math.min(
+      Math.max(
+        splittedRatio[0] / splittedRatio[1],
+        MIN_CUSTOM_DESKTOP_ONE_PANEL_ASPECT_RATIO
+      ),
+      MAX_CUSTOM_DESKTOP_ONE_PANEL_ASPECT_RATIO
+    );
+    setImportantStyles(document.querySelector(':root'), {
```

Comment:
It looks like this might not be working consistently:

https://user-images.githubusercontent.com/3860311/220702893-bdd15f36-0503-44e1-801c-e46293aa4bf4.mp4

Reply:
Should be fixed now. Thanks for the catch!

The issue was that the aspect ratio is updated only in the `storyContentLoaded` event handler, which would not be executed when the active story is already loaded before becoming active.